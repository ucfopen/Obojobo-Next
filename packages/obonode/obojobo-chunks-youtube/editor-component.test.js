import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import YouTube from './editor-component'
import { Transforms } from 'slate'
import ModalUtil from 'obojobo-document-engine/src/scripts/common/util/modal-util'

jest.mock('obojobo-document-engine/src/scripts/common/util/modal-util')
jest.mock('obojobo-document-engine/src/scripts/common/util/insert-dom-tag', () => () => {
	// simulate loading the youtube iframe api
	global.window.onYouTubeIframeAPIReady()
})
jest.mock('slate')
jest.mock('slate-react')
jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/with-slate-wrapper',
	() => item => item
)
jest.mock(
	'obojobo-document-engine/src/scripts/oboeditor/components/node/editor-component',
	() => props => <div>{props.children}</div>
)
jest.mock('obojobo-document-engine/src/scripts/common/util/uuid', () => () => 'mockId')

describe('YouTube Editor Node', () => {
	beforeEach(() => {
		jest.restoreAllMocks()
		jest.resetAllMocks()
	})

	test('YouTube component', () => {
		const component = renderer.create(<YouTube element={{ content: { videoId: 'gJ390e5sjHk' } }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('YouTube renders with no id correctly', () => {
		const component = renderer.create(<YouTube element={{ content: {} }} />)
		expect(component.toJSON()).toMatchSnapshot()
	})

	test('YouTube component handles tab', () => {
		const component = mount(<YouTube element={{ content: { videoId: 'gJ390e5sjHk' } }} selected />)

		component
			.find('button')
			.at(0)
			.simulate('keyDown', { key: 'k' })
		component
			.find('button')
			.at(0)
			.simulate('keyDown', { key: 'Tab', shiftKey: true })

		component
			.find('button')
			.at(1)
			.simulate('keyDown', { key: 'k' })
		component
			.find('button')
			.at(1)
			.simulate('keyDown', { key: 'Tab' })

		const tree = component.html()
		expect(tree).toMatchInlineSnapshot(
			`"<div><div contenteditable=\\"false\\" class=\\"obojobo-draft--chunks--you-tube viewer pad  is-selected\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center delete-button\\" contenteditable=\\"false\\"><button class=\\"button\\" contenteditable=\\"false\\">Ã—</button></div><div id=\\"obojobo-draft--chunks-you-tube-player-mockId\\" class=\\"obojobo-draft--chunks-you-tube-player\\"></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center edit-button\\" contenteditable=\\"false\\"><button class=\\"button\\" tabindex=\\"0\\" contenteditable=\\"false\\">Edit</button></div></div></div>"`
		)
	})

	test('YouTube component deletes self', () => {
		const component = mount(<YouTube element={{ content: { videoId: 'gJ390e5sjHk' } }} />)

		component
			.find('button')
			.at(0)
			.simulate('click')

		expect(Transforms.removeNodes).toHaveBeenCalled()
	})

	test('YouTube component edits properties', () => {
		const component = mount(<YouTube element={{ content: { videoId: 'gJ390e5sjHk' } }} />)

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(ModalUtil.show).toHaveBeenCalled()
	})

	test('changeProperties sets the nodes content', () => {
		const component = mount(<YouTube element={{ content: { videoId: 'gJ390e5sjHk' } }} />)

		component.instance().handleSourceChange({ mockProperties: 'mock value' })

		expect(Transforms.setNodes).toHaveBeenCalled()
	})
})
