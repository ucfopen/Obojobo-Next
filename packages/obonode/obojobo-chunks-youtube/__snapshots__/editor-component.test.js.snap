// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTube Editor Node YouTube component 1`] = `
<div>
  <div
    className="obojobo-draft--chunks--you-tube viewer pad  is-not-selected"
    contentEditable={false}
    onClick={[Function]}
  >
    <div
      className="obojobo-draft--components--button is-not-dangerous align-center delete-button"
      contentEditable={false}
    >
      <button
        className="button"
        contentEditable={false}
        onClick={[Function]}
      >
        ×
      </button>
    </div>
    <div
      className="obojobo-draft--chunks-you-tube-player"
      id="obojobo-draft--chunks-you-tube-player-mockId"
    />
    <div
      className="obojobo-draft--components--button is-not-dangerous align-center edit-button"
      contentEditable={false}
    >
      <button
        className="button"
        contentEditable={false}
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;

exports[`YouTube Editor Node YouTube renders with no id correctly 1`] = `
<div>
  <div
    className="obojobo-draft--chunks--you-tube viewer pad  is-not-selected"
    contentEditable={false}
    onClick={[Function]}
  >
    <div
      className="obojobo-draft--components--button is-not-dangerous align-center delete-button"
      contentEditable={false}
    >
      <button
        className="button"
        contentEditable={false}
        onClick={[Function]}
      >
        ×
      </button>
    </div>
    <div
      className="empty-frame"
    >
      <div>
        <p>
          No video specified. Click 'Edit' to add one.
        </p>
      </div>
    </div>
    <div
      className="obojobo-draft--components--button is-not-dangerous align-center edit-button"
      contentEditable={false}
    >
      <button
        className="button"
        contentEditable={false}
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;
